### PR TITLE
[0780] Sort by training provider in search results not working

### PR DESCRIPTION
**Pull request by defong**
### Context

Sort by training provider in search results not working

### Changes proposed in this pull request
Fixed js issues
Amended course ordering

### Guidance to review

image

When sorting by training provider ascending
Then sort by provider ascending (a to z)
and sort by course name ascending (a to z)

When sorting by training provider descending
Then sort by provider descending (z to a)
and sort by course name ascending (a to z)

See 
https://find-pr-1600.london.cloudapps.digital/results?age_group=secondary&fulltime=false&hasvacancies=true&l=2&parttime=false&qualifications%5B%5D=QtsOnly&qualifications%5B%5D=PgdePgceWithQts&qualifications%5B%5D=Other&senCourses=false&subject_codes%5B%5D=08&subject_codes%5B%5D=11&sortby=1

Uses 
https://github.com/DFE-Digital/publish-teacher-training/pull/3128

### Checklist

- [x] Rebased `main`
- [x] Cleaned commit history
- [x] Tested by running locally

### CODE REVIEW

**Comment by defong** [comment]
[🐞 Sort by training provider in search results not working](https://trello.com/c/omD9igbG/780-%F0%9F%90%9E-sort-by-training-provider-in-search-results-not-working)

**Review by simonwhatley — APPROVED** [review]

File: app/webpacker/scripts/sort-by.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
+const initSortBy = ({sortby_form_id, sortby_form_select_class}) => {
+  const sortby_form = document.getElementById(sortby_form_id);
+  if (!sortby_form) return;
+
+  const sort_by_form_select = sortby_form.getElementsByClassName(sortby_form_select_class)[0];
+
```
Comment (darokel) [inline app/webpacker/scripts/sort-by.js:6]: Can we reduce the space here and bottom of the file

File: app/webpacker/scripts/sort-by.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+  try {
+    if(sort_by_form_select) {
+      sort_by_form_select.onchange = () => {
+        console.log("asasd")
```
Comment (darokel) [inline app/webpacker/scripts/sort-by.js:11]: Could we remove

File: app/webpacker/scripts/sort-by.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+        sortby_form.submit();}
```
Comment (darokel) [inline app/webpacker/scripts/sort-by.js:12]: Closing curly bracket should be on new line

File: app/webpacker/scripts/sort-by.js [hunk 4]
```diff
@@ -0,0 +1,24 @@
+const initSortBy = ({sortby_form_id, sortby_form_select_class}) => {
+  const sortby_form = document.getElementById(sortby_form_id);
```
[the remark this answers is not in the document]
Comment (defong) [reply app/webpacker/scripts/sort-by.js]: Done

File: app/webpacker/scripts/sort-by.js (hunk 2 above)
Comment (defong) [reply app/webpacker/scripts/sort-by.js]: Done

File: app/webpacker/scripts/sort-by.js (hunk 1 above)
Comment (defong) [reply app/webpacker/scripts/sort-by.js]: Done

**Review by darokel — APPROVED** [review]

**Review by victoriawinlow — APPROVED** [review]
All good, thanks Sau

**Pull request merged**